js/app/src/lite/custom-element/index.test.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it, expect, vi } from "vitest";
+import { bootstrap_custom_element } from "./index";
+
+const delay = (ms?: number) =>
+	new Promise((resolve) => setTimeout(resolve, ms));
+
+describe("bootstrap_custom_element", () => {
+	const create = vi.fn();
+
+	bootstrap_custom_element(create);
+
+	it("parses a <gradio-lite> element that contains a string literal as a direct child", async () => {
+		document.body.innerHTML = `
+<gradio-lite>
+  import gradio as gr
+</gradio-lite>
+`;
+		await delay(); // Wait for the requestAnimationFrame to run
+
+		expect(create).toHaveBeenCalledWith(
+			expect.objectContaining({
+				code: expect.stringMatching(/import gradio as gr/),
+				requirements: [],
+				files: undefined
+			})
+		);
+	});
+
+	it("parses a <gradio-lite> element that contains <gradio-code> and <gradio-requirements> elements", async () => {
+		document.body.innerHTML = `
+<gradio-lite>
+    <gradio-code>
+import gradio as gr
+    </gradio-code>
+
+    <gradio-requirements>
+numpy
+scipy
+    </gradio-requirements>
+</gradio-lite>
+`;
+		await delay(); // Wait for the requestAnimationFrame to run
+
+		expect(create).toHaveBeenCalledWith(
+			expect.objectContaining({
+				code: expect.stringMatching(/import gradio as gr/),
+				requirements: ["numpy", "scipy"],
+				files: undefined
+			})
+		);
+	});
+
+	it("parses a <gradio-lite> element that contains <gradio-file> and <gradio-requirements> elements", async () => {
+		document.body.innerHTML = `
+<gradio-lite>
+    <gradio-file name="app.py" entrypoint>
+import gradio as gr
+
+from foo import foo
+    </gradio-file>
+
+    <gradio-file name="foo.py">
+def foo():
+    return "bar"
+    </gradio-file>
+
+    <gradio-requirements>
+numpy
+scipy
+    </gradio-requirements>
+</gradio-lite>
+`;
+		await delay(); // Wait for the requestAnimationFrame to run
+
+		expect(create).toHaveBeenCalledWith(
+			expect.objectContaining({
+				files: {
+					"app.py": {
+						data: expect.stringMatching(/import gradio as gr/)
+					},
+					"foo.py": {
+						data: expect.stringMatching(/def foo\(\):/)
+					}
+				},
+				entrypoint: "app.py",
+				code: undefined,
```

Comment:
This assertion is expected to fail (and to pass after merging the current `main`) but it's already passed.
Will see why.

Reply:
Ah, the CI test workflow checked out the [_test_ merge commit](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request) from this PR into `main`, so the fixed code on the `main` was tested on the CI which was expected to be successful correctly.